app/javascript/gobierto_data/lib/factories/datasets.js

```diff
@@ -12,8 +12,18 @@ const headers = {
 // Dataset-endpoint factory to get/post/put/delete API data
 export const DatasetFactoryMixin = {
   methods: {
-    getDatasetMetadata(id) {
-      return axios.get(`${endPoint}/${id}/meta`, { headers });
+    async getDatasetMetadata(id) {
+      return axios.get(`${endPoint}/${id}/meta`, { headers })
+        .then(response => {
+          return response
+        })
+        // eslint-disable-next-line no-unused-vars
+        .catch(error => {
+          if (error.response.status === 404) {
+            // eslint-disable-next-line no-unused-vars
+            this.$router.push('/datos/').catch(error => {})
+          }
+        })
```

Comment:
Why must a `factory` know the error response calls a router action? You're mixing unrelated concepts. A couple of thing here:

1. `getDatasetMetadata` shouldn't be async. It only returns a promise
2. `.then(response => { return response })` REALLY? 😒 Remove that clause
3. No catches here. Handle the catch error where `getDatasetMetadata` is being called, wrapping it into a try-catch block.